gobblin-data-management/src/main/java/gobblin/data/management/retention/dataset/FsCleanableHelper.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+package gobblin.data.management.retention.dataset;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Properties;
+import java.util.Set;
+
+import lombok.Getter;
+
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.slf4j.Logger;
+
+import com.typesafe.config.Config;
+
+import gobblin.data.management.trash.ProxiedTrash;
+import gobblin.data.management.trash.TrashFactory;
+import gobblin.data.management.version.FileSystemDatasetVersion;
+import gobblin.dataset.FileSystemDataset;
+import gobblin.util.PathUtils;
+
+/**
+ * A helper class to delete {@link Path}s of a FileSystemDatasetVersion.
+ * <p>
+ * Supports the following job level settings:
+ * <ul>
+ * <li> Simulate Mode - Only log paths to be deleted without deleting data by setting {@value #SIMULATE_KEY} to true.
+ * <li> Skip Trash - Delete permanent by setting {@value #SKIP_TRASH_KEY} to true.
+ * <li> Auto delete empty parent directories - By setting {@value #DELETE_EMPTY_DIRECTORIES_KEY} to true.
+ * <li> Proxy as owner and delete - By setting {@value #DELETE_AS_OWNER_KEY} to true.
+ * </ul>
+ * </p>
+ */
+public class FsCleanableHelper {
+
+  public static final String CONFIGURATION_KEY_PREFIX = "gobblin.retention.";
+
+  public static final String SIMULATE_KEY = CONFIGURATION_KEY_PREFIX + "simulate";
```

Comment:
They are moved from ```CleanableDatasetBase```. But I have not removed them in ```CleanableDatasetBase``` for backward compatibility.